src/test/java/org/opentripplanner/middleware/controllers/api/OtpUserControllerTest.java

```diff
@@ -0,0 +1,128 @@
+        otpUser.phoneNumber = INITIAL_PHONE_NUMBER;
+        otpUser.isPhoneNumberVerified = true;
+        Persistence.otpUsers.create(otpUser);
+    }
+
+    /**
+     * Delete the users if they were not already deleted during the test script.
+     */
+    @AfterAll
+    public static void tearDown() {
+        assumeTrue(testsShouldRun());
+        otpUser = Persistence.otpUsers.getById(otpUser.id);
+        if (otpUser != null) otpUser.delete();
+    }
+
+    /**
+     * Check that a request with a malformed number
+     * results in a 400-bad request response, and that the user phone number is unchanged.
+     */
+    @Test
+    public void invalidNumbersShouldProduceBadRequest() {
+        assumeTrue(testsShouldRun());
+        final String encodedNumber = "(555%5D%20555%2C0123"; // (555] 555,0123 encoded by JavaScript.
+
+        // 1. Request verification SMS.
+        // The invalid number should fail the call.
+        HttpResponse<String> response = mockAuthenticatedRequest(
+            String.format("api/secure/user/%s/verify_sms/%s",
+                otpUser.id,
+                encodedNumber
+            ),
+            otpUser,
+            HttpUtils.REQUEST_METHOD.GET
+        );
+        assertEquals(HttpStatus.BAD_REQUEST_400, response.statusCode());
+
+        // 2. Fetch the newly-created user.
+        // The phone number should not be updated.
+        HttpResponse<String> otpUserWithPhoneRequest = mockAuthenticatedRequest(
+            String.format("api/secure/user/%s", otpUser.id),
+            otpUser,
+            HttpUtils.REQUEST_METHOD.GET
+        );
+        assertEquals(HttpStatus.OK_200, otpUserWithPhoneRequest.statusCode());
+
+        OtpUser otpUserWithPhone = JsonUtils.getPOJOFromJSON(otpUserWithPhoneRequest.body(), OtpUser.class);
+        assertEquals(INITIAL_PHONE_NUMBER, otpUserWithPhone.phoneNumber);
+        assertTrue(otpUserWithPhone.isPhoneNumberVerified);
+    }
+
+    @ParameterizedTest
+    @MethodSource("createPhoneNumberTestCases")
+    public void isPhoneNumberValid(Map.Entry<String, Boolean> testCase) {
+        assumeTrue(testsShouldRun());
+        String number = testCase.getKey();
+        boolean isValid = testCase.getValue();
+
+        assertEquals(isValid, NotificationUtils.isPhoneNumberIsValid(number));
+    }
+
+    private static Set<Map.Entry<String, Boolean>> createPhoneNumberTestCases() {
+        HashMap<String, Boolean> cases = new HashMap<>();
+        cases.put("5555550123", true);
+        cases.put("(555) 555-0123", true);
+        cases.put("555 555 0123", true);
+        cases.put("(555) 555,0123", false);
+        cases.put("555555", false);
+        cases.put("55555555555555", false);
+
+        return cases.entrySet();
+    }
+
+    @Test
+    public void getE164Number() {
+        assumeTrue(testsShouldRun());
+        String formattedNumber = "(800) 555-0123";
+        assertEquals("+18005550123", NotificationUtils.getRawPhoneNumber(formattedNumber));
+    }
```

Comment:
These 2 tests are for methods entirely contained within the NotificationUtils class, so they should be placed in a the NotificationUtilsTest class or it should be created if it doesn't already exist.

Reply:
I moved the check to `OtpUserController` in https://github.com/ibi-group/otp-middleware/pull/83/commits/a9386a6e2f46bc9529e14d51ff9e51106ec87ba6